### PR TITLE
add ignore section to config which will exclude analysis from those providers

### PR DESCRIPTION
**Pull request by grapigeau**

### CODE REVIEW

File: cerberus-cli/src/main/java/ca/on/oicr/gsi/cerberus/cli/Configuration.java [hunk 1]
```diff
@@ -1,11 +1,13 @@
 package ca.on.oicr.gsi.cerberus.cli;
 
+import java.util.ArrayList;
 import java.util.Map;
 
 public final class Configuration {
 
   private Map<String, PineryConfiguration> pinery;
   private Map<String, String> vidarr;
+  private ArrayList<String> ignore;
```
Comment (callunity) [inline cerberus-cli/src/main/java/ca/on/oicr/gsi/cerberus/cli/Configuration.java]: Should be `List` here, with the more specific `ArrayList` used when instantiating.

**Review by callunity — COMMENTED** [review]
Please also update the README

File: README.md [hunk 2]
```diff
@@ -32,6 +35,9 @@ The `"vidarr"` section describes all the Vidarr instances that should be used
 as file sources. The keys are the _internal name_ of that Vidarr instance and
 the value is the URL of that instance.
 
+The `"ignore"` section contains all the LIMS data sources which are present 
+in the vidarr instances but should NOT be merged.
```
Comment (callunity) [inline README.md]: What do you think of:

The `"ignore"` section contains all the LIMS provider names which are present 
in the vidarr instances' external keys but should NOT be merged when building
file provenance. If a Vidarr workflow run contains a single external key with data
from one of these `ignore` providers, the entire workflow run will be excluded.

**Review by callunity — APPROVED** [review]

File: cerberus-core/src/main/java/ca/on/oicr/gsi/cerberus/vidarr/VidarrWorkflowRunSource.java [hunk 3]
```diff
@@ -107,16 +112,23 @@ public UpdateResult<ProvenanceWorkflowRun<ExternalKey>> update() throws Exceptio
       EPOCH.labels(baseUrl).set(body.getEpoch());
       TIMESTAMP.labels(baseUrl).set(body.getTimestamp());
       ERROR.labels(baseUrl).set(0);
-      for (final var workflowRun : body.getResults()) {
-        workflowRun.setInstanceName(instanceName);
-      }
+      ArrayList<ProvenanceWorkflowRun<ExternalKey>> validBodyResults = new ArrayList<>();
```
Comment (mlaszloffy) [inline cerberus-core/src/main/java/ca/on/oicr/gsi/cerberus/vidarr/VidarrWorkflowRunSource.java]: ArrayList --> List too?

Comment (callunity) [reply cerberus-core/src/main/java/ca/on/oicr/gsi/cerberus/vidarr/VidarrWorkflowRunSource.java]: I believe it's correct as is - `new` has to refer to a class, not an interface

Comment (mlaszloffy) [reply cerberus-core/src/main/java/ca/on/oicr/gsi/cerberus/vidarr/VidarrWorkflowRunSource.java]: ArrayList for the assignment type/lhs.

Comment (grapigeau) [reply cerberus-core/src/main/java/ca/on/oicr/gsi/cerberus/vidarr/VidarrWorkflowRunSource.java]: I don't think you can have ArrayList just on the left. Do you mean change it to this?
`List<ProvenanceWorkflowRun<ExternalKey>> validBodyResults = new ArrayList<>();`

Comment (mlaszloffy) [reply cerberus-core/src/main/java/ca/on/oicr/gsi/cerberus/vidarr/VidarrWorkflowRunSource.java]: Yes, `List<...> = new ArrayList<>()` is what I meant.

File: cerberus-core/src/main/java/ca/on/oicr/gsi/cerberus/vidarr/VidarrWorkflowRunSource.java [hunk 4]
```diff
@@ -107,16 +112,23 @@ public UpdateResult<ProvenanceWorkflowRun<ExternalKey>> update() throws Exceptio
...
-        workflowRun.setInstanceName(instanceName);
-      }
+      List<ProvenanceWorkflowRun<ExternalKey>> validBodyResults = new ArrayList<>();
+      body.getResults().stream()
+          .filter(
+              workflowRun ->
+                  workflowRun.getExternalKeys().stream()
+                      .noneMatch(
+                          externalKey -> ignoreProviders.contains(externalKey.getProvider())))
+          .peek(workflowRun -> workflowRun.setInstanceName(instanceName))
```
Comment (callunity) [inline cerberus-core/src/main/java/ca/on/oicr/gsi/cerberus/vidarr/VidarrWorkflowRunSource.java]: Why did you choose to use `peek` here?

Comment (grapigeau) [reply cerberus-core/src/main/java/ca/on/oicr/gsi/cerberus/vidarr/VidarrWorkflowRunSource.java]: Originally I had ` .map(workflowRun -> { workflowRun.setInstanceName(instanceName); return workflowRun; })` and intellij suggested to "replace with peek". 
From a google looks like that's bad practice, so I will change

**Review by mlaszloffy — APPROVED** [review]

**Pull request merged**